app/frontend/wallet/wallet.ts

```diff
@@ -0,0 +1,113 @@
+import {Account} from './account'
+import BlockchainExplorer from './blockchain-explorer'
+
+const Wallet = ({config, cryptoProvider}) => {
+  const blockchainExplorer = BlockchainExplorer(config)
+
+  const accounts: Array<ReturnType<typeof Account>> = []
+
+  function discoverNewAccount() {
+    const newAccount = Account({
+      config,
+      cryptoProvider,
+      blockchainExplorer,
+      accountIndex: accounts.length,
+    })
+    accounts.push(newAccount)
+    return newAccount
+  }
+
+  async function discoverAccounts() {
+    let shouldExplore = true
+    let accountIndex = 0
+    while (shouldExplore) {
+      const newAccount = accounts[accountIndex] || discoverNewAccount()
+      const isAccountUsed = await newAccount.isAccountUsed()
+
+      shouldExplore = isAccountUsed && config.shouldExportPubKeyBulk && accounts.length < 100
```

Comment:
the number 100 should be extracted to a constant which may be used also in the UI e.g. to ensure that we don't show a dummy tile of an account to be explored, if the index happens to be beyond this `MAX_ALLOWED_ACCOUNT_INDEX`